ansible/roles/java-base/tasks/main.yml

```diff
@@ -4,8 +4,7 @@
 # installs java
 #
 
-- name: install java
-  package: name="{{ java_package_name }}" state=present
 # if this fails you want to check in vars/main.yml and add package name
 # as appropriate -- try to use generic os family if available.
-  failed_when: "'{{ java_package_name }}' == 'missing'"
+- name: install java
+  package: name="{{ java_package_name }}" state=present
```

Comment:
If the package can't be installed (because the package name was not appropriate), ansible-palybook fails immediately, at least for Ubuntu 14.04.

With the `failed_when` it shows a warning instead of a error, effectively ignoring the error. The warning is that we should not use `{{}}` expressions with `failed_when`.